libs/application/templates/announcement-of-death/src/lib/announcementOfDeathTemplate.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  ApplicationTemplate,
+  ApplicationContext,
+  ApplicationStateSchema,
+  ApplicationTypes,
+  ApplicationRole,
+  DefaultStateLifeCycle,
+  Application,
+  DefaultEvents,
+  EphemeralStateLifeCycle,
+} from '@island.is/application/core'
+import { Events, States, Roles } from './constants'
+import { dataSchema } from './dataSchema'
+import { m } from '../lib/messages'
+import { ApiActions } from './constants'
+import { Features } from '@island.is/feature-flags'
+
+const AnnouncementOfDeathTemplate: ApplicationTemplate<
+  ApplicationContext,
+  ApplicationStateSchema<Events>,
+  Events
+> = {
+  type: ApplicationTypes.ANNOUNCEMENT_OF_DEATH,
+  name: m.applicationTitle,
+  institution: m.applicationInstitution,
+  dataSchema: dataSchema,
+  readyForProduction: false,
+  featureFlag: Features.announcementOfDeath,
+  stateMachineConfig: {
+    initial: States.PREREQUISITES,
+    states: {
+      [States.PREREQUISITES]: {
+        id: 'prerequisite',
+        exit: [],
+        meta: {
+          name: 'Prerequisites',
+          actionCard: {
+            title: m.applicationTitle,
+          },
+          onEntry: {
+            apiModuleAction: ApiActions.syslumennOnEntry,
+            shouldPersistToExternalData: true,
+            throwOnError: false,
+          },
+          progress: 0.25,
+          lifecycle: EphemeralStateLifeCycle,
+          roles: [
+            {
+              id: Roles.APPLICANT,
+              formLoader: () =>
+                import('../forms/prerequisite').then((val) =>
+                  Promise.resolve(val.prerequisite()),
+                ),
+              actions: [
+                {
+                  event: DefaultEvents.SUBMIT,
+                  name: 'Submit',
+                  type: 'primary',
+                },
+                { event: DefaultEvents.REJECT, name: 'Reject', type: 'reject' },
+              ],
+              write: 'all',
+              delete: true,
+            },
+          ],
+        },
+        on: {
+          [DefaultEvents.SUBMIT]: { target: States.DRAFT },
+          [DefaultEvents.REJECT]: { target: States.DELEGATED },
+        },
+      },
+      [States.DRAFT]: {
+        meta: {
+          name: 'Draft',
+          actionCard: {
+            title: m.applicationTitle,
+          },
+          progress: 0.5,
+          lifecycle: DefaultStateLifeCycle,
```

Comment:
Default lifecycle is shouldBePruned : false. Is that correct should this draft live forever?

Reply:
Ah, good point. I'll discuss the proper lifecycle with my team

Reply:
Discussed with project manager and added proper lifecycles.

Reply:
Draft will live for 6months until pruned.